types/layer.go

```diff
@@ -359,16 +367,42 @@ func requireImportHash(imports Imports) error {
 }
 
 func getImportFromInterface(v interface{}) (Import, error) {
+	var hash, dest string
+	var mode fs.FileMode
+	var uid, gid int
+
 	m, ok := v.(map[interface{}]interface{})
-	var hash string
 	if ok {
 		// check for nil hash so that we won't end up with "nil" string values
 		if m["hash"] == nil {
 			hash = ""
 		} else {
 			hash = fmt.Sprintf("%v", m["hash"])
 		}
-		return Import{Hash: hash, Path: fmt.Sprintf("%v", m["path"])}, nil
+
+		if m["dest"] != nil {
+			if !filepath.IsAbs(m["dest"].(string)) {
+				return Import{}, errors.Errorf("Dest path cannot be relative for: %#v", v)
+			}
+
+			dest = fmt.Sprintf("%v", m["dest"])
```

Comment:
(i'm not sure, but) should this be dest = fmt.Sprintf("%q", m["dest"]) to ensure a properly quoted path string?